include/boost/histogram/python/axis_ostream.hpp

```diff
@@ -34,18 +34,10 @@ inline const char *axis_suffix(const axis::transform::log &) { return "_log"; }
 inline const char *axis_suffix(const axis::transform::sqrt &) { return "_sqrt"; }
 inline const char *axis_suffix(const axis::transform::pow &) { return "_pow"; }
 
-template <class OStream, class T>
-void stream_metadata(OStream &os, const T &t) {
-    detail::static_if<detail::is_streamable<T>>(
-        [&os](const auto &t) {
-            std::ostringstream oss;
-            oss << t;
-            if(!oss.str().empty()) {
-                os << ", metadata=" << std::quoted(oss.str());
-            }
-        },
-        [&os](const auto &) { os << ", metadata=" << detail::type_name<T>(); },
-        t);
+template <class OStream>
+void stream_metadata(OStream &os, const metadata_t &t) {
+    if(!t.is_none())
+        os << ", metadata=" << std::quoted(py::cast<std::string>(t));
```

Comment:
Do all python types cast to strings (I assume through repr or str)?

Reply:
Casting to string calls `str(obj)`, this is closest to what you hand before. We may want to call `repr(obj)`.

Reply:
The `repr` streaming needs to be improved anyway. For `repr`, we want to show all axis options and the metadata even if it is none, but we don't show these things when they are the default values. That's ok for `str` but `repr` should be more specific.

Reply:
This is another topic for another PR